### PR TITLE
[Tests] fix sharding tests

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

So the following tests pass:

```
tests/models/unets/test_models_unet_2d.py::NCSNppModelTests::test_sharded_checkpoints
tests/models/autoencoders/test_models_vae.py::ConsistencyDecoderVAETests::test_sharded_checkpoints
```

Some comments in line.

### CODE REVIEW

File: tests/models/test_modeling_common.py [hunk 1]
```diff
@@ -905,11 +905,13 @@ def test_sharded_checkpoints(self):
             actual_num_shards = len([file for file in os.listdir(tmp_dir) if file.endswith(".safetensors")])
             self.assertTrue(actual_num_shards == expected_num_shards)
 
-            new_model = self.model_class.from_pretrained(tmp_dir)
+            new_model = self.model_class.from_pretrained(tmp_dir).eval()
             new_model = new_model.to(torch_device)
 
             torch.manual_seed(0)
+            _, inputs_dict = self.prepare_init_args_and_inputs_for_common()
```
Comment (sayakpaul) [inline tests/models/test_modeling_common.py:912]: Otherwise the `generator` (which is required as an input when using the `ConsistencyDecoderVAE` will be consumed and its states will change in-place. So, we need to re-initialize here.

File: src/diffusers/models/embeddings.py [hunk 2]
```diff
@@ -415,9 +415,10 @@ def __init__(
 
         if set_W_to_weight:
             # to delete later
+            del self.weight
             self.W = nn.Parameter(torch.randn(embedding_size) * scale, requires_grad=False)
-
             self.weight = self.W
+            del self.W
```
Comment (sayakpaul) [inline src/diffusers/models/embeddings.py:421]: Necessary to mitigate the "tied weight tensors" error from `safetensors`.

File: tests/models/autoencoders/test_models_vae.py [hunk 3]
```diff
@@ -361,9 +361,10 @@ class ConsistencyDecoderVAETests(ModelTesterMixin, unittest.TestCase):
     forward_requires_fresh_args = True
 
     def inputs_dict(self, seed=None):
-        generator = torch.Generator("cpu")
-        if seed is not None:
-            generator.manual_seed(0)
```
Comment (sayakpaul) [inline tests/models/autoencoders/test_models_vae.py:366]: This was likely a mistake because if a `seed` is provided you would want to use that to do: `generator.manual_seed(seed)` and not `generator.manual_seed(0)` using a fixed `seed`.

**Review by SunMarc — APPROVED** [review]
Thanks for fixing @sayakpaul !

**Review by DN6 — APPROVED** [review]

File: src/diffusers/models/embeddings.py (hunk 2 above)
Comment (townwish4git) [reply src/diffusers/models/embeddings.py:421]: @sayakpaul curious about codes here, may i ask for more details if possible? ;)
